tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/effects/debugger_effects.ts

```diff
@@ -435,11 +432,17 @@ export class DebuggerEffects {
     }>
   ): Observable<void> {
     return prevStream$.pipe(
-      filter(({begin, end}) => end > begin),
-      tap(() => {
-        this.store.dispatch(executionDigestsRequested());
+      withLatestFrom(this.store.select(getExecutionDigestsLoaded)),
+      filter(([{begin, end}, loaded]) => {
+        return (
+          end > begin &&
+          findBeginEndRangeIndex(loaded.loadingRanges, begin, end) === -1
```

Comment:
Btw, you may want smarter heuristic than this: 

if `loadingRanges` was [0, 100] and new begin and end were <0, 5>, you don't have to make the request again. Of course, the converse does not hold.

Reply:
Good point. This is relatively minor because the case you mentioned happens only at the end of the eager-execution timeline. But it's easy to implement: a simple change to the `findBeginEndRangeIndex()` suffices. So I did it.